Formula/readline.rb

```diff
@@ -68,17 +68,11 @@ class Readline < Formula
   uses_from_macos "ncurses"
 
   def install
-    args = ["--prefix=#{prefix}"]
-    args << "--with-curses" if OS.linux?
-    system "./configure", *args
-
-    args = []
-    args << "SHLIB_LIBS=-lcurses" if OS.linux?
-    # There is no termcap.pc in the base system, so we have to comment out
-    # the corresponding Requires.private line.
-    # Otherwise, pkg-config will consider the readline module unusable.
-    inreplace "readline.pc", /^(Requires.private: .*)$/, "# \\1"
-    system "make", "install", *args
+    system "./configure", "--prefix=#{prefix}", "--with-curses"
+    # TODO: Work out why we need to set `SHLIB_LIBS` at all.
+    #       This is needed only on Linux, and is a no-op on macOS.
+    #       Arch does something similar, but Debian doesn't.
+    system "make", "install", "SHLIB_LIBS=-lncurses"
```

Comment:
@iMichka, do you remember why you added `SHLIB_LIBS` in #65524? From reading a bit it seems like it shouldn't be needed 😅

Reply:
Actually, I suspect you needed it because of the `inreplace`, maybe.

Reply:
The only reason I see why I would have done that was that there was a build error, and this was the only workaround to fix it. We wanted to use brewed `ncurses`,  as asked by `--with-curses`. But the build would fail to find ncurses, and `SHLIB_LIBS` was the only way I found to make the compilation pass.

See [https://bugzilla.redhat.com/show_bug.cgi?id=162023 for example.](https://lists.gnu.org/archive/html/bug-readline/2017-10/msg00007.html)

Reply:
Builds fine without it though: #111811